src/DefaultUnleash.php

```diff
@@ -113,6 +128,18 @@ public function getVariant(string $featureName, ?Context $context = null, ?Varia
         $variant = $this->variantHandler->selectVariant($feature, $context);
         if ($variant !== null) {
             $this->metricsHandler->handleMetrics($feature, true, $variant);
+
+            if (method_exists($feature, 'hasImpressionData') && $feature->hasImpressionData()) {
+                $event = new ImpressionDataEvent(
+                    ImpressionDataEventType::GET_VARIANT,
+                    Uuid::v4(),
+                    clone $this->configuration,
+                    clone $context,
+                    clone $feature,
+                    clone $variant,
```

Comment:
Same as before, cloning the objects instead of providing the actual instances.

Reply:
Definitely agree